docker/sds-api/spine-directory-service/sds/request/tests/test_cpm.py

```diff
@@ -280,3 +280,172 @@ def test_translated_device_data_multiple_devices(self):
+        dir_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), os.path.join("test_data", "cpm", "returned_endpoints.json"))
+        incoming_json = self._read_file(dir_path)
+        filters = [
+            
+            {
+                "org_code": "",
+                "service_id": "urn:nhs:names:services:ebs:PRSC_IN070000UK08",
+            },
+            {
+                "org_code": "RTX",
+                "service_id": "",
+            },
+            {
+                "org_code": "",
+                "party_key": "RTX-821088"
+            },
+            {
+                "service_id": "",
+                "party_key": "RTX-821088"
+            },
+            {
+                "service_id": "urn:nhs:names:services:ebs:PRSC_IN070000UK08",
+                "party_key": ""
+            },
+            {
+                "org_code": "",
+                "service_id": "",
+            },
+            {
+                "org_code": "",
+                "party_key": "",
+            },
+            {
+                "service_id": "",
+                "party_key": ""
+            },
+            {
+            },
+            {
+                "service_id": "urn:nhs:names:services:ebs:PRSC_IN070000UK08",
+            },
+            {
+                "org_code": "RTX",
+            },
+            {
+                "party_key": "RTX-821088"
+            },
+            {
+                "org_code": "",
+                "service_id": "urn:nhs:names:services:ebs:PRSC_IN070000UK08",
+                "party_key": ""
+            },
+            {
+                "org_code": "RTX",
+                "service_id": "",
+                "party_key": ""
+            },
+            {
+                "org_code": "",
+                "service_id": "",
+                "party_key": "RTX-821088"
+            },
+            {
+                "org_code": "",
+                "service_id": "",
+                "party_key": ""
+            }
+        ]
+        for filt in filters:
+            with self.assertRaises(tornado.web.HTTPError) as context:
+                filter_cpm_endpoints_response(incoming_json, filt)
+            raised_exception = context.exception
+            self.assertEqual(raised_exception.status_code, 400)
+            self.assertEqual(raised_exception.log_message, 'Missing or invalid query parameters. Should one of following combinations: [\'organization=https://fhir.nhs.uk/Id/ods-organization-code|value&identifier=https://fhir.nhs.uk/Id/nhsServiceInteractionId|value&identifier=https://fhir.nhs.uk/Id/nhsMhsPartyKey|value\'\'organization=https://fhir.nhs.uk/Id/ods-organization-code|value&identifier=https://fhir.nhs.uk/Id/nhsServiceInteractionId|value\'\'organization=https://fhir.nhs.uk/Id/ods-organization-code|value&identifier=https://fhir.nhs.uk/Id/nhsMhsPartyKey|value\'\'identifier=https://fhir.nhs.uk/Id/nhsServiceInteractionId|value&identifier=https://fhir.nhs.uk/Id/nhsMhsPartyKey|value\']')
```

Comment:
This is just a question - but which parameters are missing, I'm struggling to tell from the error message assertion

Reply:
Discussed with James and this is just a generic error message that covers all cases